### ISSUE TITLE
Canceled Document should lead to reverse GL Entries, not deletion

### ISSUE THREAD

**Issue by chdecultot** [frappe/erpnext#11130]
Hi,

Currently when we cancel an invoice (for example) in ERPNext, the GL entries associated to this document are simply deleted.

Normally a reverse entry should be posted instead of deleting the existing entry (which should never happen in accounting).

Initial Entry:
**Customer - Debit 50
Income - Credit 50**

In case of cancellation, the system should post:

**Customer - Credit 50
Income - Debit 50**

**Comment by ckosiegbu** [comment]
I think that the problem with this is that cancellation does not lead to the generation of a new document number and therefore there's no document to which to link the entries representing the reversal. Perhaps cancellation, just like resubmitting an amended document, should result in a new document.

**Comment by jk2081** [comment]
I think this should be a checkbox setting. Either delete or reverse entries.

**Comment by Olawale1** [comment]
I agree with @chdecultot and @ckosiegbu ...this approach will also improve audit trail

Kind regards,

**Comment by jwrober** [comment]
I think this depends on the planned workflow. For example if the cancellation is for an immediate amendment, then I would not want the reverse GL entry. I would want to re-link any other documents related to the PINV or SINV (such as PE) so they can be amended as well.

**Comment by ankush** [comment]
V13 does this by default now.
